Libraries/LibJS/Interpreter.cpp

```diff
@@ -47,8 +47,47 @@ Interpreter::~Interpreter()
 {
 }
 
+Value Interpreter::destructive_run(Statement& statement, ArgumentVector arguments, ScopeType scope_type)
+{
+    m_returned = false;
```

Comment:
What happens with this flag if we nest multiple calls to `destructive_run()`?

Reply:
Nothing interesting, I don't think.
`destructive_run` is meant for generators, and we don't support `yield*`, so a generator cannot *both* return and yield in the same expression, in the same pass.